route/build/src/write_simoutput.f90

```diff
@@ -133,19 +144,153 @@ end subroutine defvar
+ ierr = nf90_def_dim(ncid, trim(sTdh_DimName), ntdh, idimId)
+ if(ierr/=0)then; message=trim(message)//trim(nf90_strerror(ierr)); return; endif
+
+ ! create dimension for future UH routed flow size
+ if (routOpt==0 .or. routOpt==1)then
+   ierr = nf90_def_dim(ncid, trim(sTdh_irf_DimName), ntdh_irf, idimId)
+   if(ierr/=0)then; message=trim(message)//trim(nf90_strerror(ierr)); return; endif
+ endif
+ ! create dimension for wave number in a reach 
+ if (routOpt==0 .or. routOpt==2)then
+   ierr = nf90_def_dim(ncid, trim(sWav_DimName), nMaxWave, idimId)
+   if(ierr/=0)then; message=trim(message)//trim(nf90_strerror(ierr)); return; endif
+ endif 
+
+ ! define segment id
+ call defvar('reachID',    'reach ID',                                    '-',   (/sSeg_DimName/),                              nf90_int,    ierr, cmessage)
+ ! define time bound 
+ call defvar('time_bound', 'time bound at last time step',                'sec', (/sTB_DimName/),                               nf90_double, ierr, cmessage)
+ ! Hill-Slope 
+ call defvar('QFUTURE',    'Hill slope routed future flow',    'm3/s',(/sSeg_DimName,sTdh_DimName,sEns_dimName/),    nf90_double, ierr, cmessage)
+ call defvar('BASIN_QR',   'Hill slope routed flow',           '-',   (/sSeg_DimName,sEns_dimName/),                 nf90_double, ierr, cmessage)
+ ! IRF 
+ if (routOpt==0 .or. routOpt==1)then
+   call defvar('irfsize',    'number of future flow time step',           '-',   (/sSeg_DimName,sEns_dimName/),                 nf90_int,    ierr, cmessage)
+   call defvar('QFUTURE_IRF','IRF convoluted river flow',                 'm3/s',(/sSeg_DimName,sTdh_irf_DimName,sEns_dimName/),nf90_double, ierr, cmessage)
+ endif
+ ! KWT 
+ if (routOpt==0 .or. routOpt==2)then
+   call defvar('wavesize',   'number of wave in a reach',                 '-',   (/sSeg_DimName,sEns_dimName/),                 nf90_int,    ierr, cmessage)
+   call defvar('QF',         'flow of a wave ',                           'm3/s',(/sSeg_DimName,sWav_DimName,sEns_dimName/),    nf90_double, ierr, cmessage)
+   call defvar('QM',         'modified flow of a wave',                   'm3/s',(/sSeg_DimName,sWav_DimName,sEns_dimName/),    nf90_double, ierr, cmessage)
+   call defvar('TI',         'entry time of a wave in reach',             'sec', (/sSeg_DimName,sWav_DimName,sEns_dimName/),    nf90_double, ierr, cmessage)
+   call defvar('TR',         'time when a wave is expected to exit reach','sec', (/sSeg_DimName,sWav_DimName,sEns_dimName/),    nf90_double, ierr, cmessage)
+   call defvar('RF',         'routing flat (1 if wave has exited reach)', '-',   (/sSeg_DimName,sWav_DimName,sEns_dimName/),    nf90_int,    ierr, cmessage) 
+ endif
+ ! end definitions
+ ierr = nf90_enddef(ncid)
+ if(ierr/=0)then; message=trim(message)//trim(nf90_strerror(ierr)); return; endif
+ ! close NetCDF file
+ ierr = nf90_close(ncid)
+ if(ierr/=0)then; message=trim(message)//trim(nf90_strerror(ierr)); return; endif
```

Comment:
can you also add a few basic global attributes to the state file?

See http://cfconventions.org/Data/cf-conventions/cf-conventions-1.7/cf-conventions.html#description-of-file-contents

Reply:
Added global attributes for model outputs and restarts (contents could be modified better)